aten/src/ATen/native/mkldnn/Linear.cpp

```diff
@@ -24,22 +24,45 @@ Tensor mkldnn_linear(
 namespace at {
 namespace native {
 
+// Helper function for getting an ideep tensor out of an aten Tensor.
+// Note in case the aten Tensor is a dense tensor, make sure:
+//   1. the aten tensor is 2d, so that 3d*2d Linear is viewed as 2d*2d.
+//   2. the aten tensor is contiguous.
+inline ideep::tensor get_mkldnn_tensor(const at::Tensor& tensor) {
+  if (tensor.is_mkldnn()) {
+    return at::native::itensor_from_mkldnn(tensor);
```

Comment:
shouldn;'t we handle reshape in this case too?

Reply:
yes, the logic is updated:
1. for mkldnn input tensor. return the `ideep::tensor` if input is 2d, otherwise `reshape` input to 2d first.
2. for dense input tensor. view it as 2d if the `input.dim > 2`.

Correspondingly, mkldnn output when dim >2 is also reshaped.